File: packages/appNuxt/database/geohash_utils.ts
import { IFBRestaurant } from 'ieattatypes/types/index'

const geohash = require('ngeohash')

export const convertToGeoHash = (latitude: number, longitude: number) => {
  return geohash.encode(latitude, longitude, 12)
}

export const getGeoHashForRestaurant = (restaurant: IFBRestaurant) => {
  return geohash.encode(restaurant.latitude, restaurant.longitude, 8)
}

File: packages/appGetx/packages/app_models/lib/src/utils/geohash_utils.dart
import 'package:geohash/geohash.dart';
// import 'package:ieatta/src/appModels/models/Restaurants.dart';

/// numberOfChars of GeoHash for restaurant.
/// @type {number}
const int numberOfCharsForRestaurant = 6;

/// numberOfChars of GeoHash for photo.
/// @type {number}
const int numberOfCharsForPhoto = 6;

String convertToGeoHash(final double latitude, final double longitude) {
  return Geohash.encode(latitude, longitude, codeLength: numberOfCharsForPhoto);
}

// String getGeoHashForRestaurant(final ParseModelRestaurants restaurant) {
//   return Geohash.encode(restaurant.latitude, restaurant.longitude, codeLength: numberOfCharsForRestaurant);
// }
